src/register/RegistrationPage.jsx

```diff
@@ -36,20 +49,28 @@ const RegistrationPage = (props) => {
     backedUpFormData,
     backendValidations,
     fieldDescriptions,
+    handleInstitutionLogin,
     intl,
+    institutionLogin,
+    optionalFields,
     registrationErrorCode,
     registrationResult,
     shouldBackupState,
     submitState,
+    thirdPartyAuthApiStatus,
+    thirdPartyAuthContext,
     usernameSuggestions,
     validationApiRateLimited,
     // Actions
     backupFormState,
+    setUserPipelineDetailsLoaded,
     getRegistrationDataFromBackend,
+    userPipelineDataLoaded,
     validateFromBackend,
   } = props;
 
   const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
+  const tpaHint = useMemo(() => getTpaHint(), []);
```

Comment:
I think this will work but I have some points.

- useMemo hook is used to memoize the result of an expensive computing function so that its value is returned on every render rather than again computing. since in your suggestion, we will be passing an object and the hook will memoize the object, not the functions. and I think this will be a contradiction to what the useMemo hook is used for. although it may work.

- supposing it is working great, but in our scenario, we are not passing any dependency, that means in-case if we ever needed to put dependency for a single function like for `getTpaHint()` function, then the other two functions will be running along with the desired function failing the purpose of useMemo.

- plus it's not any efficiency problem to have more than one useMemo hook, instead, I think it's more readable.

I am more inclined towards the first point where the hook will be memoizing the object instead of function. but I still need your point of view if my points are valid or not.

Reply:
Lets discuss on huddle